### PR TITLE
Compare ent res simulations

### PR DESCRIPTION
**Pull request by matthiasto**
Allow to compare entrainment simulations
Fix blank space above figure

### CODE REVIEW

**Review by awirb — CHANGES_REQUESTED** [review]

File: avaframe/ana3AIMEC/ana3AIMEC.py [hunk 1]
```diff
@@ -399,6 +394,9 @@ def postProcessAIMECReport(rasterTransfo, pLim, newRasters, cfgPath, cfgFlags):
     # compute runout based on peak pressure
     runout, runoutMean, elevRel, deltaH = aT.computeRunOut(rasterTransfo, pLim, PPRCrossMax, PPRCrossMean, transformedDEMRasters)
 
+    if cfgFlags.getboolean('analyzeMass'):
+        releaseMass, entrainedMass, entMassArray, totalMassArray, finalMass, relativMassDiff, grIndex, grGrad, time = aT.analyzeMass(fnameMass)
```
Comment (fso42) [inline avaframe/ana3AIMEC/ana3AIMEC.py:398]: please see issue #357

File: avaframe/out3Plot/outAIMEC.py [hunk 2]
```diff
@@ -275,10 +275,10 @@ def visuMass(resAnalysis, cfgPath, cfgFlags):
     # ax2.set_ylabel('z [m]')
     ax2.spines['right'].set_color('r')
     ax2.tick_params(axis='y', colors='r')
-    ax2.plot(time, (dataMass[0, :]-dataMass[1, :])/dataMass[0, :]*100, 'r', label='Difference')
-    ax2.set_ylabel('Entrained Mass Difference relative to total mass [%]', color='r')
+    ax2.plot(time, (totalMassArray[0, :]-totalMassArray[1, :])/entMass[0]*100, 'r', label='Difference')
+    ax2.set_ylabel('Entrained Mass Difference relative to total entrained mass [%]', color='r')
 
-    axes.flatten()[1].text(time[-1]/4, (np.nanmin(dataMass[0, :])+np.nanmax(dataMass[0, :]))/2, 'Entrained Mass Difference : %.2f kg \n Relative to entrained mass : %.2f %% \n Relative to total mass : %.2f %% ' %
+    axes.flatten()[1].text(time[-1]/4, (np.nanmin(totalMassArray[0, :])+np.nanmax(totalMassArray[0, :]))/2, 'Entrained Mass Difference : %.2f kg \n Relative to entrained mass : %.2f %% \n Relative to total mass : %.2f %% ' %
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:281]:
```suggestion
    axes.flatten()[1].text(time[-1]/4, (np.nanmin(totalMassArray[0, :])+np.nanmax(totalMassArray[0, :]))/2, 
                                    'Entrained Mass Difference : %.2f kg \n Relative to entrained mass : %.2f %% \n Relative to total mass : %.2f %% ' %
```

File: avaframe/out3Plot/outAIMEC.py [hunk 3]
```diff
@@ -428,20 +432,20 @@ def visuComparison(rasterTransfo, inputs, cfgPath, cfgFlags):
     fig = plt.figure(figsize=(pU.figW*3, pU.figH*2))#, constrained_layout=True)
     ax1 = plt.subplot2grid((3,3), (0,0), rowspan=3)
 
+    y_lim = min(np.nanmax(runoutLength)+20, s[-1])
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:435]:
```suggestion
    yLim = min(np.nanmax(runoutLength)+20, s[-1])
```

File: avaframe/out3Plot/outAIMEC.py [hunk 4]
```diff
@@ -428,20 +432,20 @@ def visuComparison(rasterTransfo, inputs, cfgPath, cfgFlags):
...
     # get color map
     cmap, _, _, norm, ticks = makePalette.makeColorMap(pU.cmapPres, pLim,
                                                        np.nanmax((refDataPressure)),
                                                        continuous=pU.contCmap)
     cmap.set_under(color='w')
     ref0, im = pU.NonUnifIm(ax1, l, s, refDataPressure, 'l [m]', 's [m]',
-                         extent=[l.min(), l.max(), s.min(), s.max()],
+                         extent=[l.min(), l.max(), 0, y_lim],
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:442]:
```suggestion
                         extent=[l.min(), l.max(), 0, yLim],
```

File: avaframe/out3Plot/outAIMEC.py [hunk 5]
```diff
@@ -456,7 +460,7 @@ def visuComparison(rasterTransfo, inputs, cfgPath, cfgFlags):
     cmap.set_bad(color='w')
     elev_max = 5
     ref0, im3 = pU.NonUnifIm(ax2, l, s[indStartOfRunout:], (dataDiff), 'l [m]', 's [m]',
-                         extent=[l.min(), l.max(), s[indStartOfRunout:].min(), s[indStartOfRunout:].max()],
+                         extent=[l.min(), l.max(), s[indStartOfRunout:].min(), y_lim],
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:463]:
```suggestion
                         extent=[l.min(), l.max(), s[indStartOfRunout:].min(), yLim],
```

File: avaframe/out3Plot/outAIMEC.py [hunk 6]
```diff
@@ -578,6 +582,29 @@ def resultWrite(cfgPath, cfgSetup, rasterTransfo, resAnalysis):
...
+    fid.write('\n')
+    fid.write('{:<15s}'.format('Mean'))
+    for j in range(np.shape(output)[0]):
+            fid.write('{:<15.3f}'.format(np.nanmean(output[j][:])))
+    fid.write('\n')
+    fid.write('{:<15s}'.format('STD'))
+    for j in range(np.shape(output)[0]):
+            fid.write('{:<15.3f}'.format(np.nanstd(output[j][:])))
+    fid.write('\n')
+
```
Comment (fso42) [inline avaframe/out3Plot/outAIMEC.py:607]: Please make a small internal function for the repetitive part...

**Review by fso42 — CHANGES_REQUESTED** [review]

**Review by awirb — CHANGES_REQUESTED** [review]
* adjust yLim of plots to display all the difference data

**Pull request merged**